Source/WebCore/editing/Editing.cpp

```diff
@@ -793,7 +795,7 @@ bool isTabSpanTextNode(const Node* node)
 
 HTMLSpanElement* tabSpanNode(const Node* node)
 {
-    return isTabSpanTextNode(node) ? downcast<HTMLSpanElement>(node->parentNode()) : nullptr;
+    return isTabSpanTextNode(node) ? uncheckedDowncast<HTMLSpanElement>(node->parentNode()) : nullptr;
```

Comment:
Why `uncheckedDowncast` instead of `dynamicDowncast` here?

Reply:
The checks in `isTabSpanTextNode()` assure us that it should succeed. And we only want to convert the parent to a span when that is true, so I'm not sure how dynamicDowncast would help here.

Reply:
Can change the `is` function to return the downcasted pointer instead. `dynamicTabSpanNodeDowncast`

Reply:
That would require changes to a few files, but it doesn't seem impossible and might clarify things a bit as well. Will give that a go tomorrow.

Reply:
On reflection, this function, `tabSpanNode` **is** `dynamicTabSpanTextNodeDowncast` and so it’s about merging the two functions into one.

Reply:
Removing `const` is no big deal; it’s not super-meaningful to pass a `const Node*` when you can just call `parentNode` and get a non-const `ContainerNode*` to the parent, for example.

Not being able to add `RefPtr` seems an unrelated problem; would be nice to make progress on that safety project, but not necessary to do this at the same time.

We want to minimize the use of `uncheckedDowncast` for safety reasons, but also of `downcast` for performance reasons. But neither of those needs to hold up what we’re doing here in this patch.

Reply:
Thanks for the feedback! I will land this patch minus this `uncheckedDowncast<>` and then when I have some bandwidth again I'll put up the above (gist) patch for review.